File: frontend/src/types/job.ts
/**
 * Type definitions for Celery Background Jobs and Live Status Polling.
 */

export interface JobStatusData {
  id: string;
  project_id: string;
  project_name?: string;
  dataset_id: string;
  dataset_name?: string;
  user_id: string;
  status: 'PENDING' | 'RUNNING' | 'SUCCESS' | 'FAILED' | 'CANCELLED';
  target_column: string;
  task_type: 'classification' | 'regression';
  celery_task_id?: string;
  error_message?: string;
  started_at?: string;
  completed_at?: string;
  created_at: string;
  model_id?: string;
  metrics?: Record<string, any>;
  artifact_path?: string;
}

export interface AsyncTrainingLaunchData {
  job_id: string;
  celery_task_id?: string;
  status: string;
  message: string;
}

File: backend/app/schemas/job.py
"""Pydantic schemas for Celery Background Jobs and Status Polling."""

import uuid
from typing import Any, Dict

from pydantic import BaseModel


class JobStatusResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    project_name: str | None = None
    dataset_id: uuid.UUID
    dataset_name: str | None = None
    user_id: uuid.UUID
    status: str
    target_column: str
    task_type: str
    celery_task_id: str | None = None
    error_message: str | None = None
    started_at: str | None = None
    completed_at: str | None = None
    created_at: str
    model_id: uuid.UUID | None = None
    metrics: Dict[str, Any] | None = None
    artifact_path: str | None = None


class AsyncTrainingLaunchResponse(BaseModel):
    job_id: uuid.UUID
    status: str
    message: str
